tests/integration/relation_tests/legacy_relations/test_charm_legacy_relations.py

```diff
@@ -119,3 +125,86 @@ async def test_legacy_db_ops(ops_test: OpsTest) -> None:
+
+    # verify auth is still disabled
+    unit = ops_test.model.applications[DATABASE_APP_NAME].units[2]
+    connection = unit.public_address + ":" + str(PORT)
+    assert not await auth_enabled(
+        connection
+    ), "MongoDB requires disabled authentication to support legacy relations"
+
+
+async def test_new_relation_fails_with_legacy(ops_test: OpsTest) -> None:
+    """Verify new relation joining results in blocked when legacy relations exist.
+
+    Legacy relations disable auth, therefore new relations should be prevented from joining
+    """
+    # app charm uses new relation interface
+    app_charm = await ops_test.build_charm(NEW_APP_PATH)
+    await ops_test.model.deploy(app_charm, num_units=1, application_name=NEW_APP_NAME)
+    await ops_test.model.wait_for_idle(apps=[NEW_APP_NAME], status="active", timeout=1000)
+
+    # a new relation to mongodb while its related to legacy relation should result in failure
+    await ops_test.model.add_relation(f"{NEW_APP_NAME}:{NEW_RELATION_NAME}", DATABASE_APP_NAME)
+    await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="blocked", timeout=1000)
+    assert (
+        ops_test.model.applications[DATABASE_APP_NAME].units[0].workload_status_message
+        == "cannot have both legacy and new relations"
+    )
+
+    assert (
+        ops_test.model.applications[DATABASE_APP_NAME].units[0].workload_status == "blocked"
+    ), "unit should go into blocked state when new relation and legacy relations are both added"
+
+    # verify auth is still disabled
+    unit = ops_test.model.applications[DATABASE_APP_NAME].units[0]
+    connection = unit.public_address + ":" + str(PORT)
+    assert not await auth_enabled(
+        connection
+    ), "MongoDB requires disabled authentication to support legacy relations"
+
+
+async def test_legacy_relation_fails_with_new(ops_test: OpsTest) -> None:
+    """Verify legacy relation joining results in blocked when new relations exist."""
+    database = await ops_test.build_charm(".")
+    await ops_test.model.deploy(database, num_units=1, application_name=ANOTHER_DATABASE_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[ANOTHER_DATABASE_APP_NAME], status="active", timeout=1000
+    )
+
+    # add new relation to pre-existing application from previous test
+    await ops_test.model.add_relation(
+        f"{NEW_APP_NAME}:{SECOND_NEW_RELATION_NAME}", ANOTHER_DATABASE_APP_NAME
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[NEW_APP_NAME, ANOTHER_DATABASE_APP_NAME], status="active", timeout=1000
+    )
+
+    # add legacy relation
+    await ops_test.model.add_relation(GRAYLOG_APP_NAME, ANOTHER_DATABASE_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[ANOTHER_DATABASE_APP_NAME], status="blocked", timeout=1000
+    )
+
+    assert (
+        ops_test.model.applications[ANOTHER_DATABASE_APP_NAME].units[0].workload_status_message
+        == "cannot have both legacy and new relations"
```

Comment:
Nit, also personal preference. Might be better to avoid testing for specific status messages as these are likely to change. 

Maybe look for `isinstance`, and `workload_status_message is not None` or something.

Reply:
good point, I'll remove this, but keep the blocked check